docs/developer/design/dashboard-architecture.md

```diff
@@ -0,0 +1,660 @@
+
+```mermaid
+flowchart TD
+    ConfigTopic["Beamlime Config Kafka Topic"]
+    DataTopic["Beamlime Data Kafka Topic"]
+
+    subgraph BackendServices
+        MonitorData["monitor_data service"]
+        DetectorData["detector_data service"]
+        DataReduction["data_reduction service"]
+    end
+
+    DashboardApp["Beamlime Dashboard"]
+
+    %% Data publishing
+    MonitorData -- Publishes --> DataTopic
+    DetectorData -- Publishes --> DataTopic
+    DataReduction -- Publishes --> DataTopic
+
+    %% Data consumption by dashboard
+    DataTopic -- Feeds --> DashboardApp
+
+    %% Config flows
+    DashboardApp -- Publishes config --> ConfigTopic
+    DashboardApp -- Consumes config --> ConfigTopic
+
+    ConfigTopic -- Consumed by --> MonitorData
+    ConfigTopic -- Consumed by --> DetectorData
+    ConfigTopic -- Consumed by --> DataReduction
+
+    DataReduction -- Publishes config (workflow specs) --> ConfigTopic
+```
+
+**Key Points:**
+- Backend Services publish data streams to a single Kafka topic
+- The Dashboard consumes this data topic, feeding into internal `DataService` components
+- The Dashboard both publishes to and consumes from the config topic
+- All backend services consume the config topic for configuration updates
+- The `data_reduction` service can publish configuration messages (workflow specs, status)
+
+## High-Level Architecture
+
+```mermaid
+graph TD
+    subgraph "External Systems"
+        K1[Kafka Data Streams]
+        K2[Kafka Config Topic]
+    end
+
+    subgraph "Infrastructure Layer"
+        KB[KafkaBridge]
+        MS[MessageSource]
+    end
+
+    subgraph "Application Layer"
+        CS[ConfigService]
+        DS[DataServices]
+        PM[Param Models]
+        WC[WorkflowController]
+    end
+
+    subgraph "Presentation Layer"
+        W1[Plots]
+        UI[Workflow Widgets]
+        W2[Config Widgets]
+    end
+
+    CS <--> WC
+    DS <--> WC
+    WC --> UI
+    K1 --> MS
+    K2 <--> KB
+    MS --> DS
+    DS --> W1
+    KB --> CS
+    CS <--> PM
+    PM --> W2
+    W2 --> PM
+```
+
+### Component Overview
+
+The architecture is structured in three main layers:
+
+- **Infrastructure Layer**: Manages Kafka integration and external message sources
+- **Application Layer**: Contains business logic, orchestration, and data management
+- **Presentation Layer**: Handles GUI components and user interaction
+
+<!-- TODO: Verify if WorkflowController is implemented and how it interacts with existing components -->
+
+## Configuration Architecture - Pydantic vs Param Separation
+
+The dashboard implements a configuration system that maintains a clear separation between frontend widgets (using Param for interactive controls) and backend validation/communication (using Pydantic models throughout). The `ConfigService` operates entirely with Pydantic models, ensuring type safety and validation consistency.
+
+### Configuration Flow Overview
```

Comment:
The layers in this graph seem unrelated to the layers in the graph above. Is that true? That would indicate that the actual code is not structured in the layers outlined here.

Reply:
It is not in submodules if that is what you ask. But there is definitely a layer structure, without using specific names in the code.

Reply:
I have already fixed the confusing naming and removed some redundancy. The code is structured in classes. Conceptually it is possible to group those into layers, e.g., the GUI layer (aka presentation layer) which has classes defining widgets. There are no classes corresponding to layers if that is what you ask. But doesn't it still make sense to speak about layers?

Reply:
I also find these diagrams confusing because it's not saying the same thing as the high level architecture diagram as JL pointed out.

And I think it's too detailed for an overview. 

We can, for example, remove `MessageBridge/KafkaBridge` or combine them with the `Beamlime Config Topic`.
And `TOAEdgesParam` doesn't seem very important to specify unless it's just an example.

Can we either make it much more simple or keep it consistent with the diagram above?

There is another similar diagram in the `How-It-Works` section under `BaseParamModel Mechanism` so maybe we can keep this diagram very simple instead.

Reply:
I thought about just moving on and revisit it later as you suggested,
but it's a type of documentation that we would not bother to update/discuss once it's merged so maybe it's better to just fix it now than later....